### PR TITLE
[flutter_tools] do not error flutter doctor on missing AS/intellij plugins

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

In cases where the Intellij/AS plugins are not located, display links to where they can be downloaded but do not surface an error. Now looks like:

```
[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from
      https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from
      https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
```

This should generally reduce confusion about whether the plugins are required for _every_ installed IDE. For example, frequently users may only install AS so that they can install the Android SDK - or they may have multiple copies of Intellij installed.

 https://github.com/flutter/flutter/issues/66762

### CODE REVIEW

**Comment by jonahwilliams** [comment]
@devoncarew Do you know if the links I've included are stable?

**Comment by jonahwilliams** [comment]
@DanTup

**Review by zanderso — APPROVED** [review]
For this and the other one, is there an emoji that we could put next to the message with the extension urls?

**Comment by jonahwilliams** [comment]
@zanderso what sort of emoji were you thinking of? maybe like a little computer?

**Comment by zanderso** [comment]
It looks like we already use(d) 🔨 for something. Maybe that would be good instead of adding a new one.

File: packages/flutter_tools/lib/src/doctor.dart [hunk 1]
```diff
@@ -342,6 +343,10 @@ class Doctor {
             hangingIndent = 0;
             indent = 6;
           }
+          if (message.contextUrl != null) {
+            final String prefix = globals.terminal.supportsEmoji ? '🔨 ' : ' ';
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/doctor.dart:347]: @zanderso WDUT?

Comment (zanderso) [reply packages/flutter_tools/lib/src/doctor.dart:347]: The Logger already does the emoji support check, so you can just use it directly here.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/doctor.dart:347]: TIL

**Pull request merged**